test/validation.js

```diff
@@ -475,6 +425,69 @@ describe('validation tests', () => {
     })
   })
 
+  describe('requiresArg', () => {
+    it('fails when a required argument value of type number is missing', (done) => {
+      yargs()
+        .option('w', {type: 'number', requiresArg: true})
+        .parse('-w', (err, argv, output) => {
```

Comment:
I also changed all of these tests to use `parse(...)` instead of `fail().argv` because the `fail()` pattern takes 8 seconds to error when it fails, vs instantly in the `parse()` pattern.